### PR TITLE
Binyamin/add builtin extensions

### LINKED ISSUE #19: Add built-in extensions
It may be thought of as part of #13 in the sense that each extension type should be applied separately.

So users can do:
```py
from extype import extensions

extensions.apply()  # all
extensions.apply(list, int)  # only for int & list
```

I don't know if it's the best idea, so you should also suggest the way you think is best to do this.

### PR DESCRIPTION
**Pull request by binyamin555**
Fix #19 #21

### CODE REVIEW

**Review by elazarcoh — APPROVED** [review]

**Review by neriyaco — APPROVED** [review]

**Comment by binyamin555** [comment]
I added the dict extensions and some tests. Please review once again and confirm.

Also, this now fixes #12

**Comment by binyamin555** [comment]
fixes #21

File: src/extype/extension_utils.py [hunk 1]
```diff
@@ -13,16 +13,23 @@ class Protocol(IntFlag):
...
+def extend_type_with(typ, cls, recurse_parents=True):
     """
     Extends the given type with methods from the given class
 
     :typ: The type to extend
     :cls: The class containing the methods to add to the given type
     """
 
+    if not isinstance(typ, type):
+        raise TypeError("May only extend types")
```
Comment (elazarcoh) [inline src/extype/extension_utils.py:25]: better error message: "... type(typ)"

**Review by elazarcoh — COMMENTED** [review]
added a few not terribly important comments.
I think it's ready for initial release 🥳

File: src/extype/extension_utils.py (hunk 1 above)
Comment (binyamin555) [reply src/extype/extension_utils.py:25]: Uh, what?

File: src/extype/builtin_extensions/float_ext.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class FloatExtension:
+    """
+    A class that contains methods to extend the builtin `float` type.
+    """
+
+    @overload
+    def round(self: float) -> int: ...
+    @overload
+    def round(self: float, ndigits: int) -> Union[float, int]: ...
```
[the remark this answers is not in the document]
Comment (binyamin555) [reply src/extype/builtin_extensions/float_ext.py:19]: Python 3.7 does not have that so we'll need to do it later. also, typing doesn't work anyway...

File: src/extype/extension_utils.py (hunk 1 above)
Comment (elazarcoh) [reply src/extype/extension_utils.py:25]: the current error message can be improved, if you tell the user what type is the typ variable.

Comment (binyamin555) [reply src/extype/extension_utils.py]: Gotcha

**Pull request merged**